client/request.go

```diff
@@ -0,0 +1,93 @@
+package client
+
+import (
+	log "github.com/cihub/seelog"
+	"github.com/nu7hatch/gouuid"
+
+	"github.com/b2aio/typhon/errors"
+)
+
+type Request interface {
```

Comment:
I still don't like interfaces with just one implementation that contains mostly trivial getter methods. I get that it protects developers from messing with data, but then again we're all grown ups and don't like boilerplate.

FWIW most types I remember seeing in the go standard library don't bother to hide their internal fields behind a read-only interface (for example http.Request)